### PR TITLE
Linfa 0.6 update

### PR DESCRIPTION
**Pull request by quietlychris**
Work-in-progress for update to `linfa v0.6`, including removing `mdbook-scientific`, adding a linear regression model, and adding CI for the code.

### CODE REVIEW

**Comment by quietlychris** [comment]
I'm requesting an initial review for these changes, which updates all code to 0.6, removes `mdook-scientific`, re-structures the repository, and adds content for linear regression. I realize this should probably have been a set of smaller PRs, but one thing led to another, and i think the end result is comprehensive enough for a thumbs-up or thumbs-down approach for these changes. I also think this should make it easier for new content to be added in the future.

**Comment by bytesnake** [comment]
Hi Chris,
I'm currently on vacation and will return latest on 07/01, will review then.
Cheers

28.12.2022 04:33:10 chris m ***@***.***>:

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
-mdbook-scientific/target/*
 book/*
 *.aux
-fragments
-src/assets
+plots/*
+target/
+Cargo.lock
```
Comment (YuhanLiin) [inline .gitignore:5]: Add newline

File: examples/kmeans.rs [hunk 2]
```diff
@@ -33,7 +34,7 @@ fn main() {
     let dataset = DatasetBase::from(data);
     let rng = thread_rng(); // Random number generator
     let n_clusters = 3;
-    let model = KMeans::params_with_rng(n_clusters, rng)
+    let model = KMeans::params_with(n_clusters, rng, LInfDist)
```
Comment (YuhanLiin) [inline examples/kmeans.rs:37]: Shouldn't this be L2 distance for Euclidean distance?

File: src/5_linear_regression.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Per usual, we'll create some data using one of our built-in functions. This simply creates an `Array2<f64>` with two columns, one of which will be our x-axis and the other our y-axis. We're generating this artificially, but remember, we could get this from a real data source like processing a CSV file or reading in values from a sensor.
+
+```rust,no_run
+{{#include ../examples/linear_regression.rs:create_data}}
+```
+Now that we have the initial data, let's break that down into someting that we can use for our regression; a `data` array and a `target` array. Fortunately, this is pretty simple with the `slice()` and `column()` functions provided by `ndarray`. We're also going to want to grab the maximum values for our arrays (and round them up to the nearest integer using the `ceil()` function) to be used for plotting those values a little bit later.
```
Comment (YuhanLiin) [inline src/5_linear_regression.md:10]: someting -> something

File: src/5_linear_regression.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+```rust,no_run
+{{#include ../examples/linear_regression.rs:data_format}}
+```
+
+Once the data is formatted, we'll be able to nicely add it into the `linfa`-native `Dataset` format, along with the appropriate feature names. If you're running into funky error related to array shapes in your code, this section and the step before (where we create our `data` and `target` data structures) are ones you should double-check; dynamically-shaped arrays as found in most scientific computing libraries, Rust-based or not, can be tricky. 
+
+In fact, as you may have experienced yourself, it's very common that the pre-processing steps of many data science problems (filtering, formatting, distributing, etc.) are actually the most complicated and often where a little bit of additioanl effort can save you a lot of trouble down the road.
```
Comment (YuhanLiin) [inline src/5_linear_regression.md:18]: additioanl -> additional

**Comment by YuhanLiin** [comment]
Generating plots in Rust requires a surprising amount of boilerplate.

**Comment by quietlychris** [comment]
Yeah, the boilerplate is a bit of a pain to be honest; it's actually one of the reasons that I think having a book like this is useful; it can be a little frustrating trying to figure out how to do that by yourself (the API is extensive), so I'm hoping this will cut down on the time that people need to get going. 

At some point, I'm hoping that `plotters` might consider working on a higher-level API with a bunch on default in the Builder pattern or something, combined with an `ndarray` plug-in or something, but it doesn't seem to be a high priority and I don't really have the time to design something like that myself at the moment.

File: src/5_linear_regression.md (hunk 4 above)
Comment (quietlychris) [reply src/5_linear_regression.md]: Fixed!

File: src/5_linear_regression.md (hunk 3 above)
Comment (quietlychris) [reply src/5_linear_regression.md]: Fixed!

File: examples/kmeans.rs (hunk 2 above)
Comment (quietlychris) [reply examples/kmeans.rs]: Yes, you're right; fixed!

File: .gitignore (hunk 1 above)
Comment (quietlychris) [reply .gitignore:5]: I'm not sure what this newline is for?

**Pull request closed without merging**